src/Command/CommandBase.php

```diff
@@ -71,38 +74,46 @@ abstract class CommandBase extends Command implements LoggerAwareInterface {
    *   An OutputInterface instance.
    *
    * @throws \Acquia\Cli\Exception\AcquiaCliException
+   * @throws \Exception
    */
   protected function initialize(InputInterface $input, OutputInterface $output) {
     $this->input = $input;
     $this->output = $output;
     $this->formatter = $this->getHelper('formatter');
     $this->setLogger(new ConsoleLogger($output));
+
+    /** @var TelemetryHelper $telemetry_helper */
+    $telemetry_helper = $this->getApplication()->getContainer()->get('telemetry_helper');
+    /** @var \Zumba\Amplitude\Amplitude $amplitude */
+    $amplitude = $this->getApplication()->getContainer()->get('amplitude');
+    $telemetry_helper->initializeAmplitude($amplitude, $this->getApplication()->getVersion());
     $this->questionHelper = $this->getHelper('question');
+    $this->checkTelemetryPreference();
 
     /** @var \Acquia\Cli\AcquiaCliApplication $application */
     $application = $this->getApplication();
 
-    if ($this->commandRequiresAuthentication() && !$application->isMachineAuthenticated()) {
+    /** @var \Webmozart\KeyValueStore\JsonFileStore $cloud_datastore */
+    $cloud_datastore = $this->getApplication()->getContainer()->get('cloud_datastore');
+    if ($this->commandRequiresAuthentication() && !$application::isMachineAuthenticated($cloud_datastore)) {
       throw new AcquiaCliException('This machine is not yet authenticated with Acquia Cloud. Please run `acli auth:login`');
     }
 
     $this->loadLocalProjectInfo();
-    $this->checkTelemetryPreference();
   }
 
+
   /**
    * Check if telemetry preference is set, prompt if not.
    */
-  protected function checkTelemetryPreference() {
-    $datastore = $this->getApplication()->getContainer()->get('acli_datastore');
-    $telemetry = $datastore->get(DataStoreContract::SEND_TELEMETRY);
-    if (!isset($telemetry) && $this->input->isInteractive()) {
+  public function checkTelemetryPreference(): void {
```

Comment:
I'd prefer for this to be in the TelemetryHelper class, but for some reason doing that causes the $inputs for the CommandTester to not register. Something weird about dependency injection of inputs?

Reply:
@danepowell I'd like to have a follow up PR to move this to TelemetryHelper. We can debug there.